src/main/java/core/basesyntax/service/impl/FruitListImpl.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.FruitTransaction;
+import core.basesyntax.service.FruitList;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FruitListImpl implements FruitList {
+    @Override
+    public List<FruitTransaction> getFruitList(List<FruitTransaction> list) {
+        List<String> listOfFruitsString = list.stream()
+                .map(FruitTransaction::getFruit)
+                .distinct()
+                .toList();
+        List<FruitTransaction> listOfFruitsAsObjects = new ArrayList<>();
+        for (String fruit : listOfFruitsString) {
+            int defaultQuantity = 0;
```

Comment:
The variable 'defaultQuantity' is redundant as it's always set to 0 and used immediately in the next line. You can directly use 0 in the constructor of 'FruitTransaction' on line 18.